plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java

```diff
@@ -158,26 +158,34 @@ public void run() {
 		protected void runOCBinary(MultiStatus multiStatus) {
 			ConsoleUtils.registerConsoleListener(new ConsoleListener(this));
 			final MessageConsole console = showConsole();
-			final MessageConsoleStream stream = console.newMessageStream();
+			final MessageConsoleStream os = console.newMessageStream();
+			os.setEncoding("UTF-8");
+			BufferedOutputStream stream = new BufferedOutputStream(os);
 			try {
 				final InputStream logs = new BufferedInputStream(capability.getLogs(true, key.container, OpenShiftBinaryOption.SKIP_TLS_VERIFY));
-				int c;
-				while(running && (c = logs.read()) != -1){
-					if(!stream.isClosed()) {
-						stream.write(c);
-					}
+				byte [] data = new byte [1024];
+				int read = 0;
+				while(running && !os.isClosed() && (read = logs.read(data)) != -1){
+					os.write(data, 0, read);
 				}
 			} catch (OpenShiftException e) {
-				stream.println(e.getMessage());
+				OpenShiftUIActivator.getDefault().getLogger().logError(e);
+				try {
+					if(stream != null)
```

Comment:
Local variable stream could not be set to null. But os can be closed and write will fail.